### PR TITLE
BEI-26: Create the followers table

### PR DESCRIPTION
**Pull request by hizikex**
#### WHY THIS PR?
We need a model for users to be able to follow one another.

#### WHAT DOES THIS PR DO?
It creates a model for users to be able to follow and unfollow one another.

#### HOW CAN THIS PR BE TESTED?
N/A

#### THINGS YOU MAY WANT THE REVIEWER TO NOTE.
N/A

### CODE REVIEW

File: src/core/database/migrations/20230726112251-create-followers.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      updated_at: {
+        allowNull: false,
+        type: Sequelize.DATE
+      }
+    });
+  },
+  async down(queryInterface, Sequelize) {
+    await queryInterface.dropTable('followers');
+  }
+};
```
Comment (LucianaDera) [inline src/core/database/migrations/20230726112251-create-followers.js:27]: can you add an extra line

**Review by LucianaDera — CHANGES_REQUESTED** [review]

File: src/core/database/migrations/20230726112251-create-followers.js (hunk 1 above)
Comment (hizikex) [reply src/core/database/migrations/20230726112251-create-followers.js:27]: Okay

File: src/core/database/migrations/20230726112251-create-followers.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  async up(queryInterface, Sequelize) {
+    await queryInterface.createTable('followers', {
+      follower_id: {
+        type: Sequelize.INTEGER,
+        primaryKey: true
+      },
+      followee_id: {
+        type: Sequelize.INTEGER,
+        primaryKey: true
+      },
```
Comment (RICHYDOS) [inline src/core/database/migrations/20230726112251-create-followers.js:13]: You can't have two primary keys in one table

**Review by RICHYDOS — CHANGES_REQUESTED** [review]

File: src/core/database/migrations/20230726112251-create-followers.js (hunk 2 above)
Comment (hizikex) [reply src/core/database/migrations/20230726112251-create-followers.js:13]: Are you sure about this.

File: src/core/database/models/followers.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        primaryKey: true,
+        autoIncrement: true
+      },
+      followee_id: {
+        type: DataTypes.STRING(200),
+        allowNull: true
+      },
+      follower_id: {
+        type: DataTypes.STRING(200),
+        allowNull: true
```
Comment (RICHYDOS) [inline src/core/database/models/followers.ts:17]: Shouldn't they both be of type "Integer" to conform with what is in the migrations file?
Then as primary keys I don't think they can be Null

File: src/core/database/models/users.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    User.belongsToMany(models.users, {
+      through: 'followers',
+      foreignKey: 'follower_id',
+      as: 'followers',
+    });
+  
+    User.belongsToMany(models.users, {
+      through: 'followers',
+      foreignKey: 'followee_id',
+      as: 'followees',
```
Comment (LucianaDera) [inline src/core/database/models/users.ts:49]: please correct the spelling... 

as:"followers"

**Review by LucianaDera — CHANGES_REQUESTED** [review]

File: src/core/database/models/users.ts (hunk 4 above)
Comment (hizikex) [reply src/core/database/models/users.ts:49]: We have followers and followees...
Line 49 is meant to be followees not followers

**Review by RICHYDOS — APPROVED** [review]

File: src/core/database/models/users.ts (hunk 4 above)
Comment (LucianaDera) [reply src/core/database/models/users.ts:49]: following might sound better

**Review by LucianaDera — APPROVED** [review]

File: src/core/database/migrations/20230726112251-create-followers.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (hizikex) [reply src/core/database/migrations/20230726112251-create-followers.js:13]: Okay

**Pull request closed without merging**